### ISSUE TITLE
The process was interrupted after running 7/16 samples

### ISSUE THREAD

**Issue by lanathemoon** [bioinfo-biols/SEVtras#10]
Hello! First of all, thank you very much for developing the algorithm!
Due to stability issues with the hard drive, my process was interrupted after running 7/16 samples. I would like to ask if the algorithm has a way to read and continue with the "tmd_out" of the 7 samples it has already output. 

In addition, I ran SEVrecognizer on a 128GB memory computer(There was an OOM error on a 64G computer), and it takes about 18 hours to calculate a single sample. Is this a normal phenomenon?
 
Thank you very much!

**Comment by RuiqiaoHe** [comment]
Yes, you only need run SEVtras for the remaining samples. After finished, you can used following code to integrate all samples:
```
import SEVtras
from SEVtras.main import sEV_aggregator
sEV_aggregator(out_path='the path you set to output in the previous step', name_list=['the sample name1 in your list', 'the sample name2 in your list', 'the sample nameN in your list'], max_M=1000, score_t=1e-15, threads=30, search_UMI=500, flag=0)
```
Regarding runtime, it usually takes dozen minutes with multiple processes. Could you please set more threads with `predefine_threads` to accelerate SEVtras?

**Comment by lanathemoon** [comment]
Thank you for your prompt response! I've noticed that the default value for the `predefine_threads` parameter is -2, which I understand means 'nproc - 2'. My system's `nproc` is 32, so having a couple of extra threads probably won't make much of a difference, right? Do you have any other suggestions or advice you could offer? Thank you in advance!

**Comment by lanathemoon** [comment]
1) I am running this on a terminal in my virtual machine, 2) indeed I am using 10X samples. I am not quite certain about the cause at the moment, but I will further explore this matter. Thank you for your patience in responding! Additionally, I would like to inquire about the 'adata_cell' parameter in ESAI_calculator, with 'adata_cell_path="./tests/test_cell.h5ad"'. Where should I obtain the file for this parameter when using my own data? I noticed that it is provided in your test package. Thank you very much for your assistance!

**Comment by RuiqiaoHe** [comment]
The long runtime may relate to the incompatibility of python package `multiprocessing` in the virtual machine, because I only test SEVtras in a naive Linux environment. I suggest that you can test a sample in naive Linux, and it may be much faster.
And the "adata_cell" is a matrix of cells obtained by preprocessing and clustering with scanpy or other tools. This can be obtained according to the conventional single-cell processing workflow from cellranger output, for example [scanpy](https://scanpy-tutorials.readthedocs.io/en/latest/pbmc3k.html).

**Comment by lanathemoon** [comment]
Dear professor RuiqiaoHe,

I am reaching out again for some guidance on generating Extended Data Fig. 7-a and d, specifically concerning the UMAP plots. I have attempted to use the sEV_SEVtras.h5ad file from the recognizer's output, which I believe includes only sEVs. However, the resulting plot did not appear as scattered as the ones presented in the article (see attached image [777f0818f215c33457bee65ed02a6ed](https://github.com/bioinfo-biols/SEVtras/assets/157283602/c64832c2-b0c2-4b1e-a080-c1ca200aa330)). Additionally, I used the raw_SEVtras.h5ad file for another plot, and despite adjusting the resolution to 0.03, it resulted in over 200 clusters, which seems excessive (refer to image [dfab39d331576e26166219cb27d3d84](https://github.com/bioinfo-biols/SEVtras/assets/157283602/ab8ea904-cbbe-4a0d-a21e-161464ff683c)).

My concern is that the clusters are not as well-defined and appear more aggregated compared to the published figures. I am wondering if I should perform all the QC processes again, which I might have overlooked previously. Could you please advise which specific output file should be utilized to recreate the UMAP that would closely resemble the one in the publication? Any suggestions or recommendations would be greatly appreciated to improve the clarity and separation of my UMAP plots.

Thank you very much for your time and assistance.

Best regards, 
Lana

**Comment by RuiqiaoHe** [comment]
Could you please then run the command `SEVtras.ESAI_calculator`, and complete the Part II analysis of SEVtras? The output will include a pdf file, named SEVumap.pdf, which would embed sEVs and cells in a umap. The results may resemble to the published figure.

**Comment by TATABOX99** [comment]
sEV_aggregator(out_path='the path you set to output in the previous step', name_list=['the sample name1 in your list', 'the sample name2 in your list', 'the sample nameN in your list'], max_M=1000, score_t=1e-15, threads=30, search_UMI=500, flag=0)
尊敬的作者您好，我想请问一下这个代码中的后面这些参数的意义是什么呢？
